yt_dlp/extractor/hotstar.py

```diff
@@ -87,7 +87,17 @@ def _call_api_v2(self, path, video_id):
 
 class HotStarIE(HotStarBaseIE):
     IE_NAME = 'hotstar'
-    _VALID_URL = r'https?://(?:www\.)?hotstar\.com/.*(?P<id>\d{10})'
+    _VALID_URL = r'''(?x)
+                        https?://(?:www\.)?hotstar\.com(?:/in)?/
+                        (?:
+                            sports/[^/]+/[^/]+/|
+                            movies/[^/]+/|
+                            tv/[^/]+/[^/]+/[^/]+/|
+                            [^/]{3,}/|
+                            |
+                        )
```

Comment:
first one is to match `https://www.hotstar.com/can-you-not-spread-rumours/1000076273`
second is to match `http://www.hotstar.com/1000000515` as `hotstar:id` can not be used since the extractor is using a proper url to get stuff.

Reply:
It may be easier to just override the suitable method similar to this:
```py
    @classmethod
    def suitable(cls, url):
        return False if YoutubeIE.suitable(url) else super(
            YoutubeTabIE, cls).suitable(url)
```

but this should work:
```py
    _VALID_URL = r'''(?x)
                        https?://(?:www\.)?hotstar\.com(?:/in)?/(?!in/)
                        (?:
                            tv/(?:[^/?#]+/){3}|
                            (?!tv/)[^?#]+/
                        )?
                        (?P<id>\d{10})
                '''
```

Reply:
https://regex101.com/r/KU95Xp/1